server/src/main/java/org/elasticsearch/script/field/WriteField.java

```diff
@@ -180,7 +205,26 @@ public void remove() {
         if (leaf == null) {
             return;
         }
-        container.remove(leaf);
+        if (container instanceof Map<?, ?> mapContainer) {
+            mapContainer.remove(leaf);
+        } else if (container instanceof List<?> listContainer) {
+            int index = parseIndex(leaf);
+            if (index >= 0 && index < listContainer.size()) {
+                listContainer.remove(index);
+            } else {
+                setError(
+                    () -> new IllegalArgumentException(
```

Comment:
A main goal of the fields API was to allow safe operations to occur without errors.  This is a major contract change and incompatible with the goal of the API.

Reply:
Oh, I see, this is only used in the no-argument get.

The presence of error smells like two use-cases jammed together and would likely lead to unexpected errors for scripting users.

It would be worth seeing if this functionality could either be added by subclassing WriteField or factoring out common methods.